File: zy70070/src/utils/id-generator.js
const { v4: uuidv4 } = require('uuid');

const generateNo = (prefix) => {
  const timestamp = Date.now().toString();
  const random = Math.random().toString(36).substring(2, 6).toUpperCase();
  return `${prefix}-${timestamp}-${random}`;
};

const generateId = () => {
  return uuidv4();
};

module.exports = {
  generateNo,
  generateId
};

File: zy72519/utils/id_generator.py
import uuid
from datetime import datetime


def generate_id(prefix: str = "") -> str:
    timestamp = datetime.now().strftime("%Y%m%d%H%M%S")
    unique_suffix = uuid.uuid4().hex[:8]
    if prefix:
        return f"{prefix}_{timestamp}_{unique_suffix}"
    return f"{timestamp}_{unique_suffix}"


def generate_batch_id() -> str:
    return f"BATCH_{datetime.now().strftime('%Y%m%d%H%M%S')}"
